src/state/swaps/swapsStore.ts

```diff
@@ -162,20 +162,42 @@ export const swapsStore = createRainbowStore<SwapsState>(
     getRecentSwapsByChain: (chainId: ChainId) => {
       const { recentSwaps } = get();
 
-      const chainSwaps = recentSwaps.get(chainId) || [];
-      return chainSwaps.sort((a, b) => b.swappedAt - a.swappedAt);
+      return recentSwaps.get(chainId) || [];
     },
     addRecentSwap(asset) {
       const { recentSwaps, latestSwapAt } = get();
       const now = Date.now();
       const chainId = asset.chainId;
       const chainSwaps = recentSwaps.get(chainId) || [];
 
-      const updatedSwaps = [...chainSwaps, { ...asset, swappedAt: now }].slice(-3);
+      const [latestSwap] = chainSwaps;
+
+      // Check if the most recent swap is the same as the incoming asset
+      if (latestSwap && latestSwap.uniqueId === asset.uniqueId) {
```

Comment:
can you help me understand why we need to distinguish between when the most recent swap is the same as the incoming asset or not?

Reply:
When the most recent swap is the same asset, we can shortcut here and just update the swappedAt for it instead of removing and re-adding.